plugins/kotlin-as-java/src/main/kotlin/converters/KotlinCompanion.kt

```diff
@@ -0,0 +1,72 @@
+package org.jetbrains.dokka.kotlinAsJava.converters
+
+import org.jetbrains.dokka.links.Callable
+import org.jetbrains.dokka.model.*
+import org.jetbrains.dokka.model.properties.PropertyContainer
+
+private const val DEFAULT_COMPANION_NAME = "Companion"
+
+internal fun DObject?.staticFunctionsForJava(): List<DFunction> {
+    if (this == null) return emptyList()
+    return functions.filter { it.isJvmStatic }
+}
+
+/**
+ * @return properties that will be visible as static for java.
+ * See [Static fields](https://kotlinlang.org/docs/java-to-kotlin-interop.html#static-fields)
+ */
+internal fun DObject?.staticPropertiesForJava(): List<DProperty> {
+    if (this == null) return emptyList()
+    return properties.filter { it.isJvmField || it.isConst || it.isLateInit }
+}
+
+internal fun DObject?.companionInstancePropertyForJava(): DProperty? {
+    if (this == null) return null
+    if (hasNothingToRender()) return null // do not show if companion not rendered
+
+    return DProperty(
+        name = name ?: DEFAULT_COMPANION_NAME,
+        modifier = sourceSets.associateWith { JavaModifier.Final },
+        dri = dri.copy(callable = Callable(name ?: DEFAULT_COMPANION_NAME, null, emptyList())),
+        documentation = emptyMap(),
+        sources = emptyMap(),
+        visibility = sourceSets.associateWith {
+            JavaVisibility.Public
+        },
+        type = GenericTypeConstructor(dri, emptyList()),
+        setter = null,
+        getter = null,
+        sourceSets = sourceSets,
+        receiver = null,
+        generics = emptyList(),
+        expectPresentInSet = expectPresentInSet,
+        isExpectActual = false,
+        extra = PropertyContainer.withAll(sourceSets.map {
+            mapOf(it to setOf(ExtraModifiers.JavaOnlyModifiers.Static)).toAdditionalModifiers()
+        })
+    )
+}
+
+internal fun DObject?.companionAsJava(): DObject? {
+    if (this == null) return null
+    if (hasNothingToRender()) return null
+
+    return asJava(
+        excludedProps = staticPropertiesForJava(),
+        excludedFunctions = staticFunctionsForJava()
+    )
+}
```

Comment:
I would make functions `companionAsJava`, `companionInstancePropertyForJava`, `staticPropertiesForJava` and `staticFunctionsForJava` be extensions of non nullable type, i.e `DObject.companionAsJava()`, and extract the null check one level higher.

This would help get rid of multiple occurrences of `if (this == null) return emptyList()` . You'd only have to correct usages to be a safe call, i.e `companion?.companionAsJava`, and not much else from what I see

Reply:
Agree with changes for `companionAsJava`, `companionInstancePropertyForJava`.
But for both `static*` extensions, there is more than one place where I expect a list so it will require more null checks on a higher level than just one here.